bpf/features.c

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Leon Hwang */
+
+#include "vmlinux.h"
+#include "bpf/bpf_helpers.h"
+#include "bpf/bpf_core_read.h"
+
+struct pwru_features {
+	bool kprobe_happened;
+	bool has_snprintf_btf;
+} features;
+
+SEC("kprobe")
+int kprobe_fn(struct pt_regs *ctx)
+{
+	features.kprobe_happened = true;
+
+	/* Detect if bpf_snprintf_btf is available.
+	 * Since: c4d0bfb45068 ("bpf: Add bpf_snprintf_btf helper")
+	 */
+	features.has_snprintf_btf = bpf_core_enum_value_exists(enum bpf_func_id, BPF_FUNC_snprintf_btf);
+
+	return BPF_OK;
+}
```

Comment:
Can we use a similar approach to https://github.com/cilium/pwru/blob/v1.0.9/internal/pwru/utils.go#L138 for feature detection?

Reply:
I'll have a try.